doc/auto_storage.md

```diff
@@ -0,0 +1,663 @@
+
+```json
+"storage": {
+     "volume_groups": [
+        {
+            "name": "newVG",
+            "physicalVolumes": [
+                { "search": { "condition": { "property": "name", "value": "/dev/sda2" } } }
+            ],
+            "logicalVolumes": [ { "name": "data", "size": "20 GiB" } ]
+        }
+    ]
+}
+```
+
+## Partitions needed for Booting
+
+When relying on the Agama proposal (see below), there are some options to configure whether (and
+where) Agama should calculate and create the extra partitions needed for booting.
+
+If the proposal is not used, Agama will always try to calculate and create those partitions taking
+the location of the root file system as a reference. That's the same approach that AutoYaST has
+followed for years.
+
+## Using the Automatic Proposal
+
+Agama can rely on the process known as Guided Proposal to calculate all the needed partitions, LVM
+devices and file systems based on some general product settings and some user preferences. That
+mechanism can also be used as part of the profile and will be executed as a last step, after
+processing all the explicit sections that describe devices.
+
+The `guided` section conforms to the following specification.
+
+```
+Guided
+  device [TargetDevice]
+  boot [BootSettings]
+  encryption [EncryptionSettings]
+  space <'delete'|'resize'|'keep'>
+  volumes [Volumes[]]
+
+TargetDevice <string|TargetDisk|TargetNewLvm|TargetReusedLvm>
+
+TargetDisk
+  disk <string|Search>
+ 
+TargetNewLvm
+  newLvmVg <<string|Search>[]>
+
+TargetReusedLvm
+  reusedLvmVg <string|Search>
+
+BootSettings
+  configure <boolean>
+  device <string|Search>
+
+EncryptionSettings
+  password <string>
+  method <string>
+  pbkdFunction <string>
+
+Volume
+  mountPath <string>
+  mountOptions <string>
+  filesystem <string>
+  autoSize <boolean>
+  minSize <string>
+  maxSize <string>
+  snapshots <Boolean>
+  target <VolumeTarget>
+
+VolumeTarget <'default'|NewPartition|NewVg|UseDevice|UseFilesystem>
+
+NewPartition
+  newPartition <string|Search>
+  
+NewVg
+  newVg <string|Search>
+  
+UseDevice
+  device <string|Search>
+
+UseFilesystem
+  filesystem <string|Search>
+```
+
+The `device` can be specified in several ways. The simplest one is using one of the strings "disk"
+or "newLvmVg". In that case, the proposal will automatically select the first disk to be used as
+target disk or as base to create the physical volumes. For example, this will create a default
+partition-based installation on the first available disk.
+
+```json
+"storage": {
+    "guided": { "device": "disk" }
+}
+```
+
+And this will do the same, but creating a new LVM volume group on that first candidate disk.
+
+```json
+"storage": {
+    "guided": { "device": "newLvmVg" }
+}
+```
+
+It's also possible to use a search section to specify a concrete disk...
+
+```json
+"storage": {
+    "guided": {
+        "device": {
+            "disk": { "search": "/dev/sda" }
```

Comment:
Would it be easier to use aliases?

Reply:
In which sense?

Reply:
I moved the usage of `search` in `guided` to Under Discussion.